testcases/OpTestHMIHandling.py

```diff
@@ -0,0 +1,349 @@
+            l_msg = "IPMI: power cycle failed"
+            raise OpTestError(l_msg)
+        if int(self.cv_SYSTEM.sys_ipl_wait_for_working_state()):
+            l_msg = "System failed to boot host OS"
+            raise OpTestError(l_msg)
+        self.cv_LPAR.lpar_get_OS_Level()
+
+    ##
+    # @brief This function executes HMI test case based on the i_test value, Before test starts
+    #        disabling kdump service to make sure system reboots, after injecting non-recoverable errors.
+    #
+    # @param i_test @type int: this is the type of test case want to execute
+    #                          BMC_CONST.HMI_PROC_RECV_DONE: Processor recovery done
+    #                          BMC_CONST.HMI_PROC_RECV_ERROR_MASKED: proc_recv_error_masked
+    #                          BMC_CONST.HMI_MALFUNCTION_ALERT: malfunction_alert
+    #                          BMC_CONST.HMI_HYPERVISOR_RESOURCE_ERROR: hypervisor resource error
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def testHMIHandling(self, i_test):
+        l_test = i_test
+        self.test_init()
+        l_con = self.cv_SYSTEM.sys_get_ipmi_console()
+        self.cv_SYSTEM.sys_ipmi_lpar_login(l_con)
+        self.cv_SYSTEM.sys_ipmi_set_unique_prompt(l_con)
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("uname -a")
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("cat /etc/os-release")
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("service kdump status")
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("service kdump stop")
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("service kdump status")
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("cd %s/external/xscom-utils/;" % self.l_dir)
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("lscpu")
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("dmesg -D")
+        if l_test == BMC_CONST.HMI_PROC_RECV_DONE:
+            self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("dmesg -C")
+            self.test_proc_recv_done()
+        elif l_test == BMC_CONST.HMI_PROC_RECV_ERROR_MASKED:
+            self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("dmesg -C")
+            self.test_proc_recv_error_masked()
+        elif l_test == BMC_CONST.HMI_MALFUNCTION_ALERT:
+            self.test_malfunction_allert()
+        elif l_test == BMC_CONST.HMI_HYPERVISOR_RESOURCE_ERROR:
+            self.test_hyp_resource_err()
+        else:
+            l_msg = "Please provide valid test case"
+            raise OpTestError(l_msg)
+        self.cv_SYSTEM.sys_run_cmd_on_ipmi_console("lscpu")
+        return BMC_CONST.FW_SUCCESS
+
+    ##
+    # @brief This function is used to test HMI: processor recovery done
+    #
+    # @return BMC_CONST.FW_SUCCESS or raise OpTestError
+    #
+    def test_proc_recv_done(self):
+        l_chip = random.choice(self.l_dic.keys())
+        l_core = random.choice(self.l_dic[l_chip])
+        if int(l_chip) == 0:
+            l_chip = "00000000"
```

Comment:
yeah, will run a getscom -l , to get id's of chips.
Thanks for reviewing, will re-send the patch with necessary changes.